packages/runtime/src/registry.tsx

```diff
@@ -49,18 +50,25 @@ export type ComponentImplementation<T = any> = React.FC<
     mergeState: MergeState;
     subscribeMethods: SubscribeMethods;
     slotsMap: SlotsMap | undefined;
+    style?: CSSProperties;
+    data?: Record<string, any>;
   }
 >;
 
+export type TraitResult = {
+  props: {
+    data?: unknown;
+    style?: CSSProperties;
```

Comment:
Looks good to me in this patch.

In the future, maybe we can have multiple styles in the trait results? With this, we can let components declare multiple customizable style slots.

Reply:
Argree, only one style is not enough.